rfcs/text/0008_service_status.md

```diff
@@ -0,0 +1,251 @@
+      handler: RouteHandler<P, Q, B>
+    ): RouteHandler<P, Q, B>;
+  }
+}
+```
+
+# Drawbacks
+
+1. **The default behaviors and inheritance of statuses may appear to be "magic" to developers who do not read the documentation about how this works.** Compared to the legacy status mechanism, these defaults are much more opinionated and the resulting status is less explicit in plugin code compared to the legacy `mirrorPluginStatus` mechanism.
+2. **The default behaviors and inheritance may not fit real-world status very well.** If many plugins must customize their status in order to opt-out of the defaults, this would be a step backwards from the legacy mechanism.
+3. **Supporting the legacy status API and the new mechansim may be tricky.** We are removing fields from the legacy API's response in the new API and must make sure we can easily translate this, or consider making a breaking change to this API.
+
+# Alternatives
+
+We could somewhat reduce the complexity of the status inheritance by leveraging the dependencies between plugins to enable and disable plugins based on whether or not their upstream dependencies are available. This may simplify plugin code but would greatly complicate how Kibana fundamentally operates, requiring that plugins may get stopped and started multiple times within a single Kibana server process. We would be trading simplicity in one area for complexity in another.
+
+# Adoption strategy
+
+By default, most plugins would not need to do much at all. Today, very few plugins leverage the legacy status system. The majority of ones that do, simply call the `mirrorPluginStatus` utility to follow the status of the legacy elasticsearch plugin.
+
+Plugins that wish to expose more detail about their availability will easily be able to do so, including providing detailed information such as links to documentation to resolve the problem.
+
+Depending on [Drawback #3](#drawbacks), we may need to break the existing status API, however this seems unlikely. This could have a large impact on users who use this API for checking Kibana node health in load balancer configurations.
+
+# How we teach this
+
+This largely follows the same patterns we have used for other Core APIs: Observables, composable utilties, etc.
+
+This should be taught using the same channels we've leveraged for other Kibana Platform APIs: API documentation, additions to the [Migration Guide](../../src/core/MIGRATION.md) and [Migration Examples](../../src/core/MIGRATION_EXMAPLES.md).
+
+# Unresolved questions
+
+1. Are the default behaviors too perscriptive? Do we know of many plugins whose status would not fall in line with these behaviors?
+2. Should we provide a status for every core service or only the services that we expect to have a status?
+3. Should the `meta` field be more structured?
+4. Should the hierarchy of the plugins be expressed in the response from the status API and/or UI of the status page?
```

Comment:
Imho no, at least not without the status API response. Having the `summary` field explaining why a plugin is in a given state is probably sufficient.